### PR TITLE
Added search functionality by logging in via the web instead of api

### PR DESCRIPTION
**Pull request by davegermiquet**

### CODE REVIEW

**Comment by davegermiquet** [comment]
Let me know what you think or if i need to do anything so you can merge it .. or if you just want to leave yours the way it is.

**Comment by cdjc** [comment]
Good work Dave! I've added you as a collaborator. If you accept, you can merge in changes yourself (https://help.github.com/articles/permission-levels-for-a-user-account-repository/).
When I first was considering this plugin I had two approaches available: The API (as used by the RNM android app), and scraping the web page. I chose the API as it's less likely to have breaking changes, but it does have some disadvantages (no search provided, for example.). But then I never actually had a close look at the HTTP conversation between a browser and the website. I do know that I wasn't very impressed at the efficiency of the HTTP conversation the Android app has through the API - quite a few unnecessary calls.

**Comment by cdjc** [comment]
I notice in the code, in the play_web_session function, that content_id can possibly be unassigned when making the makeRequest call.

File: default.py [hunk 1]
```diff
@@ -131,14 +308,39 @@ def _add_play_item(d):
...
+    sessionInstance.readCookie()
+    url_response = sessionInstance.makeRequest(url_from_search + "#1", None, True, HEADERSFORPLAYER)
+    returnReg = re.search(".+session-contentId=\"([0-9]+)\"",url_response.text)
+    if returnReg is None:
+        returnReg = re.search(".+var contentId = ([0-9]+);.*",url_response.text)
+    if returnReg is None:
+        pass
+    else:
+        content_id = returnReg.group(1)
+    url = sessionInstance.makeRequest("https://www.rightnowmedia.org/Media/Player",{"contentId": str(content_id),'showPoster' : 'false' },True,HEADERSFORPLAYER)
```
Comment (cdjc) [inline default.py:341]: content_id might be unassigned

Comment (davegermiquet) [reply default.py:341]: Yes. I  noticed that. I made a minor change to fix that so it'll error out if the content id is missing.

**Pull request merged**